src/graphql/data-sources/cacheable-data-source.ts

```diff
@@ -52,10 +44,13 @@ export abstract class CacheableDataSource extends RESTDataSource {
     update: UpdateFunction<Value>
     maxAge?: number
   }): Promise<Value> {
-    const updateCacheEntry = () => this.setCache({ key, update })
     const cacheEntry = await this.environment.cache.get<unknown>(key)
 
-    if (O.isNone(cacheEntry)) return await updateCacheEntry()
+    if (O.isNone(cacheEntry)) {
+      const initialValue = await update()
+      await this.setValue({ key, value: initialValue })
+      return initialValue
```

Comment:
```suggestion
      if (await this.lock(key)) {
        const initialValue = await update()
        await this.setValue({ key, value: initialValue })
        return initialValue
      } else {
        while (true) {
          const cacheEntry = this.environment.cache.get(key)
          
          if (O.isSome(cacheEntry)) {
            return cacheEntry.value
          } else {
            // sleep for some time
          } 
        }
      }
```

Given there is value cached we would end in calling the update function multiple times when the same endpoint is called multiple times at the the same time. Idea (above pseudo code): The first call updates the cache and the other wait for the first to finish and return the newly stored cache value.

However this might not be needed due to the nature we handle our cache and thx to the cache worker. I also do not like the above code either... -> follow up issue?!

Reply:
Hm... Since this is such an weird edge case (i.e. listeners didn't set the value and multiple clients try to get a newly created resources at basically the same time), I'd leave it like that.